crates/transaction-emitter-lib/src/emitter/account_minter.rs

```diff
@@ -585,3 +557,151 @@ pub fn prompt_yes(prompt: &str) -> bool {
+            seed: seed.map(parse_seed),
+            mint_to_root,
+            prompt_before_spending,
+            create_secondary_source_account,
+            expected_gas_per_transfer,
+            expected_gas_per_account_create,
+        }
+    }
+}
+
+impl From<&EmitJobRequest> for BulkAccountCreationConfig {
+    fn from(req: &EmitJobRequest) -> Self {
+        Self {
+            max_submit_batch_size: DEFAULT_MAX_SUBMIT_TRANSACTION_BATCH_SIZE,
+            skip_funding_accounts: req.skip_funding_accounts,
+            seed: req.account_minter_seed,
+            mint_to_root: req.mint_to_root,
+            prompt_before_spending: req.prompt_before_spending,
+            create_secondary_source_account: !req.coordination_delay_between_instances.is_zero(),
+            expected_gas_per_transfer: req.get_expected_gas_per_transfer(),
+            expected_gas_per_account_create: req.get_expected_gas_per_account_create(),
+        }
+    }
+}
+
+pub async fn bulk_create_accounts(
+    coin_source_account: &LocalAccount,
+    txn_executor: &RestApiReliableTransactionSubmitter,
+    txn_factory: &TransactionFactory,
+    account_generator: Box<dyn LocalAccountGenerator>,
+    config: BulkAccountCreationConfig,
+    num_accounts: usize,
+    coins_per_account: u64,
+) -> Result<Vec<LocalAccount>> {
+    let source_account_manager = SourceAccountManager {
+        source_account: coin_source_account,
+        txn_executor,
+        mint_to_root: config.mint_to_root,
+        prompt_before_spending: config.prompt_before_spending,
+        txn_factory: txn_factory.clone(),
+    };
+
+    let seed = config.seed.unwrap_or_else(|| {
+        let mut rng = StdRng::from_entropy();
+        rng.gen()
+    });
+    info!(
+        "AccountMinter Seed (reuse accounts by passing into --account-minter-seed): {:?}",
+        seed
+    );
+
+    let accounts = account_generator
+        .gen_local_accounts(txn_executor, num_accounts, &mut StdRng::from_seed(seed))
+        .await?;
+    info!(
+        "Generated and fetched re-usable accounts for seed {:?}",
+        seed
+    );
+
+    let all_accounts_already_exist = accounts.iter().all(|account| account.sequence_number() > 0);
+    let send_money_gas = if all_accounts_already_exist {
+        config.expected_gas_per_transfer
+    } else {
+        config.expected_gas_per_account_create
+    };
+
+    let mut account_minter = AccountMinter::new(
+        &source_account_manager,
+        txn_factory.clone().with_max_gas_amount(send_money_gas),
+        // Wrap seed once, to not have conflicts between worker and seed accounts.
+        // We also don't want to continue from the same rng, as number of accounts will affect
+        // seed accounts.
+        StdRng::from_seed(StdRng::from_seed(seed).gen()),
+    );
+
+    if !config.skip_funding_accounts {
+        let accounts: Vec<_> = accounts.into_iter().map(Arc::new).collect();
+        account_minter
+            .create_and_fund_accounts(
```

Comment:
Are the accounts not already created in the above `account_generator.gen_local_accounts` call?